packages/mock-addon/src/utils/faker.js

```diff
@@ -119,10 +119,20 @@ export class Faker {
 
         if (matched) {
             const { response, status, delay = 0 } = matched;
+
             return new Promise((resolve) => {
                 setTimeout(() => {
                     if (typeof response === 'function') {
-                        resolve(new Response(url, status, response(request)));
+                        const data = response(request);
+
+                        let status = 200;
+                        const customStatusCode = data?.header?.status;
```

Comment:
@yormy  can you please help understand why we are passing status inside the header?